prediction/src/algorithms/classify/src/classification_model.py

```diff
@@ -0,0 +1,119 @@
+import abc
+import collections
+
+
+class ClassificationModel:
+    """
+
+    Args:
+        init_model (bool | str): whether to initialise the model.
+            If str, then the model will be loaded from the init_model
+            path.
+        pull_size (int): maximum amount of batches allowed to be stored in RAM.
+    """
+
+    @abc.abstractmethod
```

Comment:
what does `abstractmethod` do in this case? Python docs say, 

> Using this decorator requires that the class’s metaclass is ABCMeta or is derived from it. 

Also, why not use a new style class, i.e., `class ClassificationModel(object):`

Reply:
Indeed, thanks :)